app/src/main/java/com/kabouzeid/gramophone/views/TouchInterceptFrameLayout.java

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * @author Lincoln (theduffmaster)
+ * 
+ * A custom {@link FrameLayout} that intercepts touch events and decides whether to consume them or
+ * pass them on to a child {@link TouchInterceptHorizontalScrollView} and its
+ * {@link TouchInterceptTextView}.
+ *
+ * This only needs to be used if the layout containing the {@link TouchInterceptHorizontalScrollView}
+ * is clickable.
+ */
+public class TouchInterceptFrameLayout extends FrameLayout {
+
+    public static final String TAG = TouchInterceptFrameLayout.class.getSimpleName();
+
+    private static final int MAX_CLICK_DISTANCE = 5;
+
+    private TouchInterceptHorizontalScrollView scrollView;
+
+    private Rect scrollViewRect;
+    private float startX;
+    private boolean isTap;
+
+    public TouchInterceptFrameLayout(@NonNull Context context) {
+        this(context, null);
+        init();
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs) {
+        this(context, attrs, 0);
+        init();
+    }
+
+    public TouchInterceptFrameLayout(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+    private void init() {
+        scrollViewRect = new Rect();
+        setTag(TouchInterceptFrameLayout.TAG);
+    }
+
+    /**
+     * @return Returns the child {@link TouchInterceptHorizontalScrollView}.
+     */
+    public TouchInterceptHorizontalScrollView getTouchInterceptHorizontalScrollView() {
+        return (TouchInterceptHorizontalScrollView) findViewWithTag(TouchInterceptHorizontalScrollView.TAG);
+    }
+
+    /**
+     * Intercepts touch events to selectively pass the event on to its child view.
+     * It also detects where the touch was placed so that if the touch is not in the scrollview, the
+     * touch is not passed to it, avoiding the child view swallowing up the long press. ACTION_MOVE
+     * actions are cancelled here and instead passed to the child view.
+     *
+     * @param e The intercepted touch event.
+     * @return True if the MotionEvent will be intercepted (i.e. it will not be passed on to its
+     *         child, but rather to the onTouchEvent method of this view).
+     */
+    @Override
+    public boolean onInterceptTouchEvent(MotionEvent e) {
+        int x = Math.round(e.getRawX());
+        int y = Math.round(e.getRawY());
+
+        scrollView = getTouchInterceptHorizontalScrollView();
+        scrollView.getGlobalVisibleRect(scrollViewRect);
+
+        boolean touchedScrollView =
+                x > scrollViewRect.left && x < scrollViewRect.right &&
+                y > scrollViewRect.top && y < scrollViewRect.bottom;
+
+        if (scrollView.isScrollable()) {
+            switch (e.getAction() & MotionEvent.ACTION_MASK) {
+                case MotionEvent.ACTION_DOWN:
+                    if (!touchedScrollView) {
+                        scrollView.cancelPendingInputEvents();
```

Comment:
I guess I missed this one when looking it over. Unfortunately this one is required. I've implemented a API check so that if the API is under 19 it will invoke scrollView.cancelLongPress() instead. That means that below API 19 if you really mess with it you can get the background of the view to turn grey.